go/tools/gopackagesdriver/gopackagesdriver_test.go

```diff
@@ -81,6 +78,29 @@ func main() {
 	fmt.Fprintln(os.Stderr, "Subdirectory Hello World!")
 }
 		`,
+		// Explicitly disable bzlmod which is enabled by default with bazel >=7.0.
+		// Remove this setup function when the test will be launched with bzlmod and
+		// the RULES_GO_REPO_NAME_FOR_TEST will be the canonical name of this rules_go module.
+		// This is required to match the stdlib label returned by the 'bazel query --consistent_labels',
+		// otherwise the gopackagesdriver will not return any stdlib packages.
+		SetUp: func() error {
+			f, err := os.OpenFile(".bazelrc", os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
```

Comment:
Does this write into the runfiles tree? Could we pass an explicit arg to all Bazel invocations instead?

Reply:
Yes. I append a new config param in the existing `.bazelrc` file created by [go/tools/bazel_testing/bazel_testing.go#L326-L338](https://github.com/bazelbuild/rules_go/blob/master/go/tools/bazel_testing/bazel_testing.go#L326-L338).
I couldn't use ` GO_BAZEL_TEST_BAZELFLAGS` because it's only for the 'bazel build' command.

But now Bazel is able to ignore incompatible parameters for all commands, I guess we could modify `bazel_testing`  as below:

``` diff
diff --git a/go/tools/bazel_testing/bazel_testing.go b/go/tools/bazel_testing/bazel_testing.go
index a414b40c..093231a4 100644
--- a/go/tools/bazel_testing/bazel_testing.go
+++ b/go/tools/bazel_testing/bazel_testing.go
@@ -331,7 +331,7 @@ func setupWorkspace(args Args, files []string) (dir string, cleanup func() error
                fmt.Fprintf(bazelrcBuf, "startup --output_user_root=%s\n", outputUserRoot)
        }
        if flags := os.Getenv("GO_BAZEL_TEST_BAZELFLAGS"); flags != "" {
-               fmt.Fprintf(bazelrcBuf, "build %s\n", flags)
+               fmt.Fprintf(bazelrcBuf, "common %s\n", flags)
        }
        if err := os.WriteFile(bazelrcPath, bazelrcBuf.Bytes(), 0666); err != nil {
                return "", cleanup, err

````
In this way, we could use `GO_BAZEL_TEST_BAZELFLAGS` for this specific test.

Reply:
Could you send a PR to rules_go with this change? We can leave this as is here and simplify it after the next update.